src/merge.js

```diff
@@ -22,23 +22,19 @@ function mergeOne(dst, src) {
   // See if 'dst' is allowed to be mutated. If not - it's overridden with a new plain object.
   const returnValue = isObject(dst) ? dst : {};
 
-  const keys = Object.keys(src);
-  for (let i = 0; i < keys.length; i++) {
-    const key = keys[i];
-
+  return Object.keys(src).reduce((result, key) => {
```

Comment:
This `for` loop was a performance optimization.

The problem with reduce is the performance. We are creating one more object (function) and scope. Accessing variables of the upper scope is slower than accessing from own scope.
More info here: https://github.com/petkaantonov/bluebird/wiki/Optimization-killers

Please, revert the loop (and the other one). Stampit has to be fast!